sequence_processing_pipeline/GenPrepFileJob.py

```diff
@@ -1,59 +1,50 @@
+        super().__init__(run_dir,
+                         self.job_name,
+                         [seqpro_path],
+                         modules_to_load)
+
         self.sample_sheet_path = sample_sheet_path
+        self.seqpro_path = seqpro_path
+        self.modules_to_load = modules_to_load
         self.output_directory = output_directory
 
     def run(self):
-        logging.debug('generating prep-files w/seqpro')
-        # the run directory, project name, and lane for the prep.
-        # from prep.py 424: name = run_id + '.' + project + '.' + lane
-        self._system_call(f'seqpro {self.sample_sheet_path}'
-                          f'{self.output_directory}')
-
-    def _process_sample_sheet(self, sample_sheet_path):
-        sheet = KLSampleSheet(sample_sheet_path)
-        valid_sheet = validate_and_scrub_sample_sheet(sheet)
-
-        if not valid_sheet:
-            s = "Sample sheet %s is not valid." % sample_sheet_path
-            raise PipelineError(s)
-
-        header = valid_sheet.Header
-        chemistry = header['chemistry']
-        needs_adapter_trimming = ('TRUE' if
-                                  header['Assay'] == 'Metagenomics'
-                                  else 'FALSE')
-
-        bioinformatics = valid_sheet.Bioinformatics
-
-        # reorganize the data into a list of dictionaries, one for each row.
-        # the ordering of the rows will be preserved in the order of the list.
-
-        lst = []
-        for i in range(0, len(bioinformatics)):
-            lst.append({})
-
-        for key in bioinformatics:
-            my_series = bioinformatics[key]
-            for index, value in my_series.items():
-                # convert all manner of positive/negative implying strings
-                # into a true boolean value.
-                if value.strip().lower() in ['true', 'yes']:
-                    value = True
-                elif value.strip().lower() in ['false', 'no']:
-                    value = False
-                lst[index][key] = value
-
-        # human-filtering jobs are scoped by project. Each job requires
-        # particular knowledge of the project.
-        return {'chemistry': chemistry, 'projects': lst,
-                'needs_adapter_trimming': needs_adapter_trimming}
+        # seqpro usage:
+        # seqpro path/to/run_dir path/to/sample/sheet /path/to/fresh/output_dir
+        cmd = [self.seqpro_path]
+        cmd.append(self.run_dir)
+        cmd.append(self.sample_sheet_path)
+
+        # note that seqpro takes in a sample-sheet as input. Unlike other
+        # Jobs that process the sample-sheet, validate parameters, and ensure
+        # that output is segregated by project name, seqpro will be doing that
+        # for us. Hence our output directory will just be a base directory
+        # that seqpro can use to write project directories into.
+        tmp = join(self.output_directory, 'FastQC_output')
```

Comment:
This is seqpro so IMOO it's confusing that the output dir is named FastQC_output ...

Reply:
@antgonza doh! That is a mistake.